nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/probes/ProbeInfoWriter.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour Al Kotob
+ */
+
+package org.nuxeo.rest.management.probes;
+
+import static org.nuxeo.ecm.core.io.registry.reflect.Instantiations.SINGLETON;
+import static org.nuxeo.ecm.core.io.registry.reflect.Priorities.REFERENCE;
+
+import java.io.IOException;
+
+import org.nuxeo.ecm.core.io.marshallers.json.AbstractJsonWriter;
+import org.nuxeo.ecm.core.io.registry.reflect.Setup;
+import org.nuxeo.ecm.core.management.api.ProbeInfo;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+
+@Setup(mode = SINGLETON, priority = REFERENCE)
+public class ProbeInfoWriter extends AbstractJsonWriter<ProbeInfo> {
+
+    public static final String WARNINGS = "warnings";
+
+    public static final String WARNING = "warning";
+
+    public static final String NAME = "name";
+
+    public static final String VERSION = "version";
+
+    @Override
+    public void write(ProbeInfo entity, JsonGenerator jg) throws IOException {
+        jg.useDefaultPrettyPrinter();
+        jg.writeStartObject();
+
+        jg.writeStringField("shortcutName", entity.getShortcutName());
+        jg.writeStringField("success", String.valueOf(entity.getStatus().isSuccess()));
+        jg.writeStringField("result", entity.getStatus().toString());
+
+        jg.writeArrayFieldStart("history");
+
+        String value = (entity.getLastRunnedDate() == null ) ? null : entity.getLastRunnedDate().toString();
+        jg.writeStartObject();
+        jg.writeStringField("last run", value);
+        jg.writeEndObject();
+
+        value = (entity.getLastSucceedDate() == null ) ? null : entity.getLastSucceedDate().toString();
+        jg.writeStartObject();
+        jg.writeStringField("last success", value);
+        jg.writeEndObject();
+
+        value = (entity.getLastSucceedDate() == null ) ? null : entity.getLastFailedDate().toString();
+        jg.writeStartObject();
+        jg.writeStringField("last fail", value);
+        jg.writeEndObject();
+
+        jg.writeEndArray();
+        jg.writeArrayFieldStart("counts");
+
+        jg.writeStartObject();
+        jg.writeStringField("run count", String.valueOf(entity.getRunnedCount()));
```

Comment:
The REST answered looks strange to me, have we specified it?

Currently we have something like:
```
{
  "counts": [
    { "run count": "..." },
    { "success count": "..." },
    { "failure count": "..." }
  ]
}
```

Why each count are in string format and not number?

Why do we use an array? It can be difficult to read data client side, for instance to get _run count_, we have a non easy way to get it (in javascript):
```
response.counts[0]['run count']
```

Something like below would be better:
```
{
  "counts": {
    "run": X,
    "success": Y,
    "failure": Z
  }
}
```

Reply:
i agree, no need to create object for each date field, create just an object (`counts` one) with this three fields

Reply:
i agree, no need to create object for each date field, create just an object (`counts` one) with this three fields